docs/guides/1230-persistent-cache.md

```diff
@@ -0,0 +1,216 @@
+:::
+
+## Persistent cache in your local filesystem
+
+To store cache in your local filesystem, you don't need much effort : just
+add `--cache-from type=local,mode=max,dest=<output folder>` to `dagger do build`.
+
+:::tip
+Using `mode=max` argument will cache **all** layers from intermediate
+steps, with is really useful in the context of Dagger where you will have
+multiple steps to execute.
+:::
+
+Here's an example that exports cache to `storage`.
+
+```shell
+dagger do build --cache-to type=local,mode=max,dest=storage 
+# ...
+
+tree storage -L 1   
+storage
+├── blobs
+├── index.json
+└── ingest
+```
+
+A new directory has been created that contains cache the run
+
+:::caution
+If you run different action in the plan, don't forget to store cache in different
+destination, so it will be not overwritten.
+:::
+
+To import the cache previously stored, you can use `--cache-from type=local,src=<cache folder>`.
+
+Here's an example, using a new buildkit daemon
+
+```shell
+# Down buildkit daemon
+docker container stop dagger-buildkitd && docker container rm dagger-buildkitd && docker volume rm dagger-buildkitd
+
+# Import cache on rebuild
+dagger do build --cache-to type=local,mode=max,dest=storage --cache-from type=local,src=storage
+[✔] actions.build.container                                                 2.3s
+[✔] client.filesystem.".".read                                              0.1s
+[✔] actions.build.container.export                                          0.0s
+[✔] client.filesystem."./output".write                                      0.4s
+```
+
+:::info
+In this part, we have how to keep cache in a local filesystem, if you want
+to see more options on local export, look at [Buildkit cache documentation](https://github.com/moby/buildkit#local-directory-1)
+:::
+
+## Persistent cache in a remote registry
+
+Buildkit can also import/export cache to a registry, it's a great way to share cache between your team and avoid
+flooding your filesystem.
+
+:::tip
+Using a registry as cache storage is more efficient than local storage because Buildkit will only re-export
+missing layers on multiple runs.
+:::
+
+Let's first deploy a simple registry in your localhost
+
+```shell
+docker run -d -p 6000:5000 --restart=always --name cache-registry registry:2
+```
+
+Then it's not much different from local export
+
+```shell
+dagger do build --cache-to type=registry,mode=max,ref=localhost:5000/cache --cache-from type=registry,ref=localhost:5000/cache
+[✔] actions.build.container                                                 1.3s
+[✔] client.filesystem.".".read                                              0.0s
+[✔] actions.build.container.export                                          0.0s
+[✔] client.filesystem."./output".write                                      0.1s
+```
+
+:::info
+See more options on registry export at [Buildkit cache documentation](https://github.com/moby/buildkit#registry-push-image-and-cache-separately)
```

Comment:
When linking to the external buildkit repo, maybe we should use a permalink in case a future update to their README causes our doc's link to no longer work: https://github.com/moby/buildkit/blob/ed5f454d2021c2378229ac68c05cf560cfe25f36/README.md#registry-push-image-and-cache-separately

The downside though is that we need to remember to update the link if Buildkit has a new release w/ updated options here, so not sure, just a thought.

Reply:
You are right! I think it's better to update the link ourselves that having broken links

Reply:
Fixed in https://github.com/dagger/dagger/pull/2035/commits/d916466adfe0a5cb53d851ba7e04033716be0afd